src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -380,44 +380,71 @@ internal static bool HasThisConstructorInitializer(this MethodSymbol method)
                         }
                     }
                 }
             }
 
+            initializerSyntax = null;
             return false;
         }
 
         internal static bool IncludeFieldInitializersInBody(this MethodSymbol methodSymbol)
         {
             return methodSymbol.IsConstructor()
-                && !methodSymbol.HasThisConstructorInitializer()
+                && !(methodSymbol.HasThisConstructorInitializer(out var initializerSyntax) && !isDefaultValueTypeConstructor(methodSymbol.ContainingType, initializerSyntax))
                 && !(methodSymbol is SynthesizedRecordCopyCtor) // A record copy constructor is special, regular initializers are not supposed to be executed by it.
                 && !Binder.IsUserDefinedRecordCopyConstructor(methodSymbol);
+
+            // A struct constructor that calls ": this()" will need to include field initializers if the
+            // parameterless constructor is a synthesized default constructor that is not emitted.
+            static bool isDefaultValueTypeConstructor(NamedTypeSymbol containingType, ConstructorInitializerSyntax initializerSyntax)
+            {
+                if (initializerSyntax.ArgumentList.Arguments.Count > 0)
+                {
+                    return false;
+                }
+                var constructor = containingType.InstanceConstructors.SingleOrDefault(m => m.ParameterCount == 0);
+                return constructor?.IsDefaultValueTypeConstructor(requireNoInitializers: true) == true;
+            }
         }
 
         /// <summary>
-        /// NOTE: every struct has a public parameterless constructor either used-defined or default one
+        /// NOTE: every struct has a public parameterless constructor either user-defined or default one
         /// </summary>
         internal static bool IsParameterlessConstructor(this MethodSymbol method)
         {
             return method.MethodKind == MethodKind.Constructor && method.ParameterCount == 0;
         }
 
         /// <summary>
-        /// default zero-init constructor symbol is added to a struct when it does not define 
-        /// its own parameterless public constructor.
-        /// We do not emit this constructor and do not call it 
+        /// Returns true if the method is the default zero-init constructor synthesized for struct types.
+        /// If the struct type is from metadata, the default constructor is synthesized when there
+        /// is no accessible parameterless constructor. If the struct type is from source, a parameterless
+        /// constructor is synthesized if there is no explicit parameterless constructor, but the synthesized
+        /// constructor is only considered the default zero-init constructor if the synthesized constructor
+        /// would not be emitted. (The default constructor will be emitted if the struct type has field
+        /// initializers and no explicit constructors.)
         /// </summary>
```

Comment:
Would `requireNoInitializers = true` make more sense as the default? That corresponds to old behavior of "default value type constructor == zero-init constructor" and seems the most common usage. If so, could be renamed to `allow...` instead of `require...`.

nit: Consider xml doc for `requireNoInitializers` parameter. Consider renaming to `requireZeroInit` to better align to comment.

Reply:
Removed the default parameter value, renamed the parameter to `requireZeroInit`, and updated comment to mention `requireZeroInit`.